Assignment2-Social Network Graph.Rmd

```diff
@@ -0,0 +1,145 @@
+---
+title: "Assignment2-Social Network Graph"
+author: "Aidi"
+date: "10/21/2018"
+output: html_document
+---
+
+## Part 1
+# Discipline graph
+```{r}
+library(igraph)
+library(dplyr)
+library(tidyr)
+
+# load the data file
+D1<-read.csv("discipline-data.csv", header=TRUE)
+D1$stid<-as.factor(D1$stid)
+
+# create from-to and vertex attributes
+# create from-to EDGE
+D2<-select(D1, tid, stid)
+EDGE<-count(D2,tid,stid)
+names(EDGE)<-c("from","to","count")
+
+# create vertex
+t1<-select(D1, tid, t.gender, t.expertise)
+t1<-unique(t1)
+t1$group<-"teacher"
+names(t1)<-c("ID","gender","study.area","group")
+
+s1<-select(D1, stid, s.gender, s.major)
+s1<-unique(s1)
+s1$group<-"student"
+names(s1)<-c("ID","gender","study.area","group")
+
+VERTEX<-bind_rows(t1, s1)
+
+# create a graph object
+g1<-graph.data.frame(EDGE, vertices=VERTEX, directed=TRUE)
+
+# create the network graph
+plot(g1, layout=layout.fruchterman.reingold, edge.arrow.size=0.3)
+
+plot(g1, layout=layout.fruchterman.reingold, edge.arrow.size=0.3, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+```
+
+## Part 2
+# Sizing vertices according to disciplinary action
+
+```{r}
+# add a new attribute in VERTEX
+t2<-EDGE%>%select(from,count)%>%group_by(from)%>%summarize(sum(count))
+names(t2)<-c("ID","count")
+s2<-EDGE%>%select(to,count)%>%group_by(to)%>%summarize(sum(count))
+names(s2)<-c("ID","count")
+INFO<-bind_rows(t2,s2)
+
+VERTEX<-full_join(VERTEX, INFO, by="ID")
+
+g2<-graph.data.frame(EDGE, vertices=VERTEX, directed=TRUE)
+plot(g2, layout=layout.fruchterman.reingold, edge.arrow.size=0.3,edge.width=EDGE$count, vertex.color=VERTEX$gender, vertex.size=VERTEX$count, vertex.shapes=as.factor(VERTEX$group))
```

Comment:
You may want to use new lines and indentations after the commas to make this code easier to read.